File: LeetCode/Easy/0551-student-attendance-record-i/0551-student-attendance-record-i.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        A = 0
        L = 0
        for v in s:
            if v == "A":
                A += 1
                L = 0
                if A > 1:
                    return False
            elif v == "L":
                L += 1
                if L > 2:
                    return False
            else:
                L = 0
        return True
                 

File: LeetCode/Easy/0551-student-attendance-record-i/0551-student-attendance-record-i.java
class Solution {
    public boolean checkRecord(String s) {
        int L = 0;
        int A = 0;
        for(char v : s.toCharArray()){
            if (v == 'L') {
                if (++L > 2) return false;
            }
            else if (v == 'A'){
                if (++A > 1) return false;
                L = 0;
            } else {L = 0;}
        }
        return true;
    }
}